web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -1218,10 +1247,19 @@ fixmystreet.display = {
             // wrong (but what is shown is correct).
             $('.js-back-to-report-list').attr('href', fixmystreet.original.href);
 
-            // Problems nearby should act the same as 'Back to all reports' on around,
-            // but on /my and /reports should go to that around page.
-            if (fixmystreet.original.page == 'around') {
-                $sideReport.find('#key-tool-problems-nearby').addClass('js-back-to-report-list');
+            // Problems nearby on /my should go to the around page,
+            // otherwise show reports within the current map view.
+            if (fixmystreet.original.page === 'around' || fixmystreet.original.page === 'reports') {
+                $sideReport.find('#key-tool-problems-nearby').click(function(e) {
```

Comment:
Need a meta/ctrl key check here.

Reply:
In fact, this function is still the same as the click handler for `'.js-back-to-report-list`, just presumably without the map state setting (if I've read it right). So might be better to factor those together?
(Also, do you still need to set `initial:true` so that the relevant popstate code is run if this page is later gone Back to; would also be fixed by factoring the two together.)

Reply:
Well spotted, that's made things simpler.